fabrics.c

```diff
@@ -505,25 +559,29 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 		goto out_free;
 	}
 
-	if (device) {
+	if (device && !force) {
 		c = nvme_scan_ctrl(r, device);
 		if (c) {
 			/* Check if device matches command-line options */
-			if (!nvme_ctrl_is_discovery_ctrl(c) ||
-			    strcmp(nvme_ctrl_get_transport(c), transport) ||
-			    strcasecmp(nvme_ctrl_get_traddr(c), traddr) ||
-			    (cfg.host_traddr && nvme_ctrl_get_host_traddr(c) &&
-			     strcmp(nvme_ctrl_get_host_traddr(c),
-				    cfg.host_traddr)) ||
-			    (cfg.host_iface && nvme_ctrl_get_host_iface(c) &&
-			     strcmp(nvme_ctrl_get_host_iface(c),
-				    cfg.host_iface)) ||
-			    (trsvcid && nvme_ctrl_get_trsvcid(c) &&
-			     strcmp(nvme_ctrl_get_trsvcid(c), trsvcid))) {
+			if (strcmp0(nvme_ctrl_get_transport(c), transport) ||
+			    strcasecmp0(nvme_ctrl_get_traddr(c), traddr) ||
+			    strcmp0(nvme_ctrl_get_host_traddr(c),
+				    cfg.host_traddr) ||
+			    strcmp0(nvme_ctrl_get_host_iface(c),
+				    cfg.host_iface) ||
+			    strcmp0(nvme_ctrl_get_trsvcid(c), trsvcid)) {
```

Comment:
Do make this a common function (eg nvmf_trargs_matchI()) which then can be shared with the discovery lookup.
Having two instances of doing the same thing always causes trouble later on when the match needs to be modified due to errors or spec updates.

Reply:
Good point. I've introduce a `ctrl_config_match()` helper for this.